### PR TITLE
:mortar_board: Topic logic gates --- Show examples of > 2 input gates

### PR DESCRIPTION
**Pull request by jameshughes89**
### What

1. Show how > 2 input gates can be made by stacking gates. 
2. Warn about nand/nor not working quite the same way
3. Mention how >2 input xor is weird 

### Why

It will come up 

### Testing

:+1: 

### Additional Notes

I originally had a large note about how in Digital you can add arbitrarily large number of inputs despite a drop down menu making it seem like there is only a max of 5. Opted to remove as it would likely be better served in a lab.

### CODE REVIEW

File: site/topics/transistors-gates/gates.rst [hunk 1]
```diff
@@ -283,6 +283,115 @@ Other Gate Symbols
...
+    * :math:`\lnot(\lnot(a \land b) \land \lnot(c \land d))`
+    * :math:`\lnot(\lnot((a \land b) \lor (c \land d))`
+    * :math:`(a \land b) \lor (a \land b)`
+
+
+* In other words, a four input nand gate is not functionally equivalent to stacking nand gates
+* Similar problems would exist for nor
+
+* Exclusive or (xor) gates can be stacked to have the same functionality as a single xor gate with many inputs
+* However, when xor gates have more than two inputs, they effectively become an even/odd check
```
Comment (twentylemon) [inline site/topics/transistors-gates/gates.rst]: This first line is confusing. _a single xor gate with many inputs_ is what is trying to be defined, and "stacking" gates together does not help define it -- neither are well defined at this point. The second line actually defines what we're after, then it starts to make sense. Though it still isn't clear why "stacking" xor leads to such a result, or if that is desirable. The 3-ary xor, for example, is true when all inputs are true, which is counter-intuitive.

**Review by twentylemon — COMMENTED** [review]
Generally, this topic feels out of place. It feels like it needs the mathematical background first (defining n-ary boolean operators), then later construct them with gates.

**Comment by jameshughes89** [comment]
> Generally, this topic feels out of place. It feels like it needs the mathematical background first (defining n-ary boolean operators), then later construct them with gates.

The idea was to be more of a "oh, btw, more inputs is normal", but I'm all for formalising things up more. 

You think elaborating on associativity for and would be sufficient (notwithstanding your point on xor)?

File: site/topics/transistors-gates/gates.rst (hunk 1 above)
Comment (jameshughes89) [reply site/topics/transistors-gates/gates.rst]: I was thinking of making this part of a lab. Though, I don't really have a full plan for the labs yet. 

The trouble with the labs is, I have no sense of course pacing.

**Comment by twentylemon** [comment]
I'm thinking this entire section can basically be a footnote. Literally a _”oh, btw, more inputs is normal"_ note, with a small English description of n-ary operators (like _and is true only when all inputs are true_). A sentence saying n-ary xor is a parity check is then probably sufficient. 

Generally I like the idea of moving this content to a lab, for all gates. Construct n-ary gates using only binary ones, is well defined and easy to check correctness since digital seems to support both.

**Comment by jameshughes89** [comment]
> I'm thinking this entire section can basically be a footnote. Literally a _”oh, btw, more inputs is normal"_ note, with a small English description of n-ary operators (like _and is true only when all inputs are true_). A sentence saying n-ary xor is a parity check is then probably sufficient.
> 
> Generally I like the idea of moving this content to a lab, for all gates. Construct n-ary gates using only binary ones, is well defined and easy to check correctness since digital seems to support both.

Simplified section. 

Relevant lab will be created at a later date.

File: site/topics/transistors-gates/gates.rst [hunk 2]
```diff
@@ -288,105 +288,46 @@ Additional Inputs
...
-
-.. figure:: and_5_input_stacked.png
-    :width: 500 px
-    :align: center
-
-    Four and gates configured such that they perform an and operation on five inputs.
-
-
-* It is common to simplify this by drawing a single and gate with the desired number of inputs
+* An and gate with more than two inputs outputs ``1`` only when all inputs are ``1``
```
Comment (twentylemon) [inline site/topics/transistors-gates/gates.rst:293]: Should gate names be changed to AND/OR/NAND etc?

**Review by twentylemon — APPROVED** [review]

File: site/topics/transistors-gates/gates.rst (hunk 2 above)
Comment (jameshughes89) [reply site/topics/transistors-gates/gates.rst:293]: I'm going to do a PR soon where I go back and update everything with some consistent strategy.

**Pull request merged**